### PR TITLE
Merge L0Mgr into jit-engine

### PR DESCRIPTION
**Pull request by dchigarev**
This PR moves standalone L0Mgr (with no codegen) and a simple test executing SPIR-V module via it into jit-engine branch.

The PR also introduces a dependency on `LLVMSPIRVLib` inside L0 execution test: spir-v translator is required due to test module generation. Although the dependency was there before, it wasn't declared in `Tests/CMakeList.txt`.

In order to build L0Mgr, you need to:
1. Install LevelZero loader: https://github.com/oneapi-src/level-zero
2. (Optional: if you want to build L0Mgr tests) Install SPIR-V translator library: https://github.com/KhronosGroup/SPIRV-LLVM-Translator
3. Pass `-DENABLE_L0=ON` parameter to Omnisci's cmake

### CODE REVIEW

**Comment by dchigarev** [comment]
Can't assign reviewers directly, so doing that via mentioning: @kurapov-peter

File: L0Mgr/L0Mgr.cpp [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+ze_driver_handle_t L0Driver::driver() const {
+  return driver_;
+}
+
+const std::vector<std::shared_ptr<L0Device>>& L0Driver::devices() const {
+  return devices_;
+}
+
+std::vector<std::shared_ptr<L0Driver>> get_drivers() {
+  zeInit(0);
```
Comment (kurapov-peter) [inline L0Mgr/L0Mgr.cpp:68]: These calls should probably be wrapped with L0_SAFE_CALL

File: L0Mgr/L0Mgr.cpp [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+  ze_module_build_log_handle_t buildlog = nullptr;
+
+  auto status = zeModuleCreate(ctx(), device_, &desc, &handle, &buildlog);
+  if (log) {
+    size_t logSize = 0;
+    L0_SAFE_CALL(zeModuleBuildLogGetString(buildlog, &logSize, nullptr));
+    std::vector<char> strLog(logSize);
+    L0_SAFE_CALL(zeModuleBuildLogGetString(buildlog, &logSize, strLog.data()));
+    std::fstream out;
+    out.open("log.txt", std::ios::app);
```
Comment (kurapov-peter) [inline L0Mgr/L0Mgr.cpp:193]: Guess we need a better logging here.

File: L0Mgr/L0Mgr.cpp [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+      .pKernelName = name,
+  };
+  ze_kernel_handle_t handle;
+  L0_SAFE_CALL(zeKernelCreate(this->handle_, &desc, &handle));
+  return std::make_shared<L0Kernel>(handle, x, y, z);
+}
+
+L0Kernel::L0Kernel(ze_kernel_handle_t handle, uint32_t x, uint32_t y, uint32_t z)
+    : handle_(handle), group_size_({x, y, z}) {
+  zeKernelSetGroupSize(handle_, x, y, z);
```
Comment (kurapov-peter) [inline L0Mgr/L0Mgr.cpp:248]: L0_SAFE_CALL

**Review by kurapov-peter — APPROVED** [review]
LGTM, please address minor comments.

**Pull request merged**